### DISCUSSION TITLE
Add TOC to ease navigation in long reports

### DISCUSSION THREAD

**Discussion by BenjaminNSD** [datapane/datapane#69] [Ideas]
Hi. I work for the European Social Survey (ESS) data archive. The ESS is a cross-national survey in Europe. We usually create quite long data processing reports (PDF) where we report data errors like wild codes, filter errors, format error etc. I just started testing datapane and it looks really promising. This could possibly replace our old-fashioned PDF reports.

Anyway, to improve readability and navigation in datapane-reports I wished there was a table of content (TOC) similar to TOC in jupyter notebook (nb-extention) and jupyterlab3.

What's your opinion? Does anybody have experience how to add this feature?

![image](https://user-images.githubusercontent.com/20127077/105098791-e8b44300-5aaa-11eb-91ac-baa448acae9c.png)

**Comment by mands** [comment]
That's a great suggestion - I think there may be a way to generate a TOC automatically from markdown headers, Groups blocks and more - we'll look into adding it as it could help a lot for longer (text-heavy) reports.

**Reply by jmrichardson** [reply]
Hi, are there any plans to add a navigation pane to reports?

**Reply by johnmicahreid** [reply]
Hi @jmrichardson, we've noted this feedback and this feature is on our roadmap. Will update you when we start working on it!
